pkg/api/api_implementation.go

```diff
@@ -257,9 +257,11 @@ func (a defaultAPI) makeRequest(ctx context.Context, obj types.Object, body inte
 
 	// We do this right on top because this checks if the Object has a correct type which is more strictly defined than just the interface.
 	// In a perfect world this would be a compile-time check.
```

Comment:
This comment is out-of-date now. The original purpose of having "this right on top" was to check if the passed Object actually is an Object, which also means it has an Identifier attribute.

With the new Object.GetIdentifier() method we have a compile-time check for this.

What we actually check here is if the Identifier is set when required and give the Object a way to deny some operations with unsuitable Identifiers being set - please update the comment accordingly :)